### PR TITLE
Increase task result limit

### PR DESCRIPTION
**Pull request by Linary**
Change-Id: Ib2c4bdf25e4604d8bea47802e724c0f0934170fb

### CODE REVIEW

File: hugegraph-core/src/main/java/com/baidu/hugegraph/task/HugeTask.java [hunk 1]
```diff
@@ -52,7 +52,8 @@
 
     private static final Logger LOG = Log.logger(HugeTask.class);
 
-    private static final int MAX_PROPERTY_LENGTH = BytesBuffer.STRING_LEN_MAX;
+    // The max
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/task/HugeTask.java:55]: improve comment

File: hugegraph-test/src/main/java/com/baidu/hugegraph/core/TaskCoreTest.java [hunk 2]
```diff
@@ -429,16 +432,13 @@ public void testGremlinJobWithError() throws TimeoutException {
         });
 
         // Test failure task with big input
-        char[] chars = new char[65536];
-        for (int i = 0; i < chars.length; i++) {
-            chars[i] = '8';
-        }
-        String bigInput = new String(chars);
+        int length = 65536 * 1024;
```
Comment (javeme) [inline hugegraph-test/src/main/java/com/baidu/hugegraph/core/TaskCoreTest.java:435]: add test 128kb is ok

Comment (Linary) [reply hugegraph-test/src/main/java/com/baidu/hugegraph/core/TaskCoreTest.java:435]: 128kb < 64MB, will not trigger exception

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/serializer/BytesBuffer.java [hunk 3]
```diff
@@ -51,6 +51,7 @@
     public static final int CHAR_LEN = Character.BYTES;
     public static final int FLOAT_LEN = Float.BYTES;
     public static final int DOUBLE_LEN = Double.BYTES;
+    public static final int BLOB_LEN = SHORT_LEN * 1024;
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/backend/serializer/BytesBuffer.java:54]: set BLOB_LEN 4 bytes is ok

File: hugegraph-core/src/main/java/com/baidu/hugegraph/task/HugeTask.java [hunk 4]
```diff
@@ -40,19 +40,21 @@
...
 import com.baidu.hugegraph.util.Log;
+import com.baidu.hugegraph.util.StringEncoding;
 
 public class HugeTask<V> extends FutureTask<V> {
 
     private static final Logger LOG = Log.logger(HugeTask.class);
 
-    private static final int MAX_PROPERTY_LENGTH = BytesBuffer.STRING_LEN_MAX;
+    // The vertex property max length, for task input and result
+    private static final int MAX_PROPERTY_LENGTH = (int) (64 * Bytes.MB);
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/task/HugeTask.java:57]: read from BLOB_LEN_MAX

File: hugegraph-core/src/main/java/com/baidu/hugegraph/task/HugeTask.java [hunk 5]
```diff
@@ -448,12 +450,12 @@ protected void property(String key, Object value) {
 
         if (this.input != null) {
             list.add(P.INPUT);
-            list.add(this.input);
+            list.add(StringEncoding.encode(this.input));
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/task/HugeTask.java:453]: call StringEncoding.compress()

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/tx/GraphTransaction.java [hunk 6]
```diff
@@ -510,6 +551,12 @@ public void removeVertex(HugeVertex vertex) {
     }
 
     public Iterator<Vertex> queryAdjacentVertices(Iterator<Edge> edges) {
+        if(this.lazyLoadAdjacentVertex){
```
Comment (zhoney) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/backend/tx/GraphTransaction.java:554]: add empty after 'if'

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/tx/AbstractTransaction.java [hunk 7]
```diff
@@ -289,8 +289,9 @@ protected void commitOrRollback() {
                 LOG.error("Failed to rollback changes:\n {}", mutation, e2);
             }
             // Rethrow the commit exception
+            // The e.getMessage maybe too long to see key information
```
Comment (zhoney) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/backend/tx/AbstractTransaction.java:292]: The e.getMessage maybe too long to see key information, therefore use e.getCause

File: hugegraph-test/src/main/java/com/baidu/hugegraph/core/TaskCoreTest.java [hunk 8]
```diff
@@ -479,16 +477,21 @@ public void testGremlinJobAndCancel() throws TimeoutException {
                               "has exceeded the max limit 10000",
                               task3.result());
 
-        // Cancel failure task with big results (task exceeded limit 64k)
-        String bigResults = "def big='123456789'; def l=[]; " +
-                            "for (i in 1..9000) l.add(big); l;";
+        // Cancel failure task with big results (task exceeded limit 64M)
+        String bigResults = "def big='12345678'; def ol=[]; def il=[]; " +
+                            "for (i in 1..2000) " +
+                            "   for (j in 1..1000) " +
```
Comment (zhoney) [inline hugegraph-test/src/main/java/com/baidu/hugegraph/core/TaskCoreTest.java:483]: delete empty of line 483 or add empty for 484-486

File: hugegraph-core/src/main/java/com/baidu/hugegraph/config/CoreOptions.java [hunk 9]
```diff
@@ -115,6 +116,22 @@ public static synchronized CoreOptions instance() {
                     10L
             );
 
+    public static final ConfigOption<Long> TASK_INPUT_SIZE_LIMIT =
+            new ConfigOption<>(
+                    "task.input_size_limit",
+                    "The gremlin job input size limit in bytes.",
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/config/CoreOptions.java:122]: gremlin job => job

File: hugegraph-core/src/main/java/com/baidu/hugegraph/config/CoreOptions.java [hunk 10]
```diff
@@ -115,6 +116,22 @@ public static synchronized CoreOptions instance() {
...
+                    rangeInt(0L, Bytes.GB),
+                    16 * Bytes.MB
+            );
+
+    public static final ConfigOption<Long> TASK_RESULT_SIZE_LIMIT =
+            new ConfigOption<>(
+                    "task.result_size_limit",
+                    "The gremlin job result size limit in bytes.",
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/config/CoreOptions.java:130]: ditto

File: hugegraph-core/src/main/java/com/baidu/hugegraph/task/HugeTask.java [hunk 11]
```diff
@@ -526,11 +535,24 @@ protected void property(String key, Object value) {
...
+
+        HugeConfig config = graph.configuration();
+        long maxLength = BytesBuffer.STRING_LEN_MAX;
+        if (name.equals(P.unhide(P.INPUT))) {
+            maxLength = config.get(CoreOptions.TASK_INPUT_SIZE_LIMIT);
+        } else if (name.equals(P.unhide(P.RESULT))) {
+            maxLength = config.get(CoreOptions.TASK_RESULT_SIZE_LIMIT);
+        }
+
+        if (property.length() > maxLength) {
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/task/HugeTask.java:552]: check the length of compressed

**Comment by javeme** [comment]
```
[ERROR] testTaskWithFailure(com.baidu.hugegraph.core.TaskCoreTest)  Time elapsed: 0.001 s  <<< ERROR!
com.baidu.hugegraph.HugeException: Failed to update/query TaskStore: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Can't delete incomplete task '1359' in status RUNNING
	at com.baidu.hugegraph.core.TaskCoreTest.setup(TaskCoreTest.java:62)
Caused by: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Can't delete incomplete task '1359' in status RUNNING
	at com.baidu.hugegraph.core.TaskCoreTest.setup(TaskCoreTest.java:62)
Caused by: java.lang.IllegalStateException: Can't delete incomplete task '1359' in status RUNNING
[ERROR] testEphemeralJob(com.baidu.hugegraph.core.TaskCoreTest)  Time elapsed: 0.003 s  <<< ERROR!
com.baidu.hugegraph.HugeException: Failed to update/query TaskStore: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Can't delete incomplete task '1359' in status RUNNING
	at com.baidu.hugegraph.core.TaskCoreTest.setup(TaskCoreTest.java:62)
Caused by: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Can't delete incomplete task '1359' in status RUNNING
	at com.baidu.hugegraph.core.TaskCoreTest.setup(TaskCoreTest.java:62)
Caused by: java.lang.IllegalStateException: Can't delete incomplete task '1359' in status RUNNING
[ERROR] testGremlinJobWithFailure(com.baidu.hugegraph.core.TaskCoreTest)  Time elapsed: 0.001 s  <<< ERROR!
com.baidu.hugegraph.HugeException: Failed to update/query TaskStore: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Can't delete incomplete task '1359' in status RUNNING
	at com.baidu.hugegraph.core.TaskCoreTest.setup(TaskCoreTest.java:62)
Caused by: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Can't delete incomplete task '1359' in status RUNNING
	at com.baidu.hugegraph.core.TaskCoreTest.setup(TaskCoreTest.java:62)
Caused by: java.lang.IllegalStateException: Can't delete incomplete task '1359' in status RUNNING
[ERROR] testGremlinJob(com.baidu.hugegraph.core.TaskCoreTest)  Time elapsed: 0.002 s  <<< ERROR!
com.baidu.hugegraph.HugeException: Failed to update/query TaskStore: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Can't delete incomplete task '1359' in status RUNNING
	at com.baidu.hugegraph.core.TaskCoreTest.setup(TaskCoreTest.java:62)
Caused by: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Can't delete incomplete task '1359' in status RUNNING
	at com.baidu.hugegraph.core.TaskCoreTest.setup(TaskCoreTest.java:62)
Caused by: java.lang.IllegalStateException: Can't delete incomplete task '1359' in status RUNNING
```

**Pull request merged**